presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingFileSystem.java

```diff
@@ -81,22 +82,27 @@ public FSDataInputStream openFile(Path path, HiveFileContext hiveFileContext)
             // FilePath is a unique identifier for a file, however it can be a long string
             // hence using md5 hash of the file path as the identifier in the cache.
             // We don't set fileId because fileId is Alluxio specific
-            FileInfo info = new FileInfo().setFileIdentifier(md5().hashString(path.toString(), UTF_8).toString())
+            FileInfo info = new FileInfo()
                     .setPath(path.toString())
                     .setFolder(false)
                     .setLength(hiveFileContext.getFileSize().get());
+            CacheContext cacheContext = new CacheContext().setCacheIdentifier(md5().hashString(path.toString(), UTF_8).toString()) {
+                public void incrementCounter(String name, long value) {
+                    // Default implementation does nothing
+                }
+            };
             if (cacheQuotaEnabled) {
                 CacheScope scope = CacheScope.create(hiveFileContext.getCacheQuota().getIdentity());
-                info.setCacheScope(scope);
+                cacheContext.setCacheScope(scope);
                 if (hiveFileContext.getCacheQuota().getQuota().isPresent()) {
-                    info.setCacheQuota(new CacheQuota(ImmutableMap.of(scope.level(), hiveFileContext.getCacheQuota().getQuota().get().toBytes())));
+                    cacheContext.setCacheQuota(new CacheQuota(ImmutableMap.of(scope.level(), hiveFileContext.getCacheQuota().getQuota().get().toBytes())));
                 }
                 else {
-                    info.setCacheQuota(CacheQuota.UNLIMITED);
+                    cacheContext.setCacheQuota(CacheQuota.UNLIMITED);
                 }
             }
             // URIStatus is the mechanism to pass the hiveFileContext to the source filesystem
-            AlluxioURIStatus alluxioURIStatus = new AlluxioURIStatus(info, hiveFileContext);
+            AlluxioURIStatus alluxioURIStatus = new AlluxioURIStatus(info, cacheContext, hiveFileContext);
```

Comment:
Instead of having an  `AlluxioURIStatus`  extending `URIStatus` with simply `hiveFileContext` added,
we can extend `CacheContext` to be `PrestoCacheContext` here with `hiveFileContext` added.

Reply:
Hi @apc999 ,  sounds great to me.  I just modified the code as you said.  Could you pls take a second look?